### PR TITLE
Update documentation files for analyzer packages

### PR DESCRIPTION
**Pull request by mavasani**
NOTE: There are all auto-generated by running `eng\common\cibuild` locally. I'll investigate if we can execute them during local builds as well so that these changes are made as part of PRs that add/change rules.

### CODE REVIEW

File: src/Microsoft.NetCore.Analyzers/Microsoft.NetCore.Analyzers.md [hunk 1]
```diff
@@ -33,33 +33,46 @@ Sr. No. | Rule ID | Title | Category | Enabled | CodeFix | Description |
...
+62 | CA5362 | Do Not Refer Self In Serializable Class | Security | False | False | This can allow an attacker to DOS or exhaust the memory of the process. |
+63 | CA5363 | Do Not Disable Request Validation | Security | True | False | Request validation is a feature in ASP.NET that examines HTTP requests and determines whether they contain potentially dangerous content. This check adds protection from markup or code in the URL query string, cookies, or posted form values that might have been added for malicious purposes. So, it is generally desirable and should be left enabled for defense in depth. |
+64 | CA5364 | Do Not Use Deprecated Security Protocols | Security | True | False | Using a deprecated security protocol rather than the system default is risky. |
+65 | CA5365 | Do Not Disable HTTP Header Checking | Security | True | False | HTTP header checking enables encoding of the carriage return and newline characters, \r and \n, that are found in response headers. This encoding can help to avoid injection attacks that exploit an application that echoes untrusted data contained by the header. |
+66 | CA5366 |  | Security | True | False |  |
```
Comment (dotpaul) [inline src/Microsoft.NetCore.Analyzers/Microsoft.NetCore.Analyzers.md]: >   [](start = 13, length = 2)

This is weird.

Comment (mavasani) [reply src/Microsoft.NetCore.Analyzers/Microsoft.NetCore.Analyzers.md]: Yes, same thing with few more rules below. Wondering why the title couldn't be accessed off the descriptor: https://github.com/dotnet/roslyn-analyzers/blob/master/src/GenerateAnalyzerRulesets/Program.cs#L395

Comment (mavasani) [reply src/Microsoft.NetCore.Analyzers/Microsoft.NetCore.Analyzers.md]: Seems like the case only for analyzers deriving from `UseXmlReaderBase`

Comment (mavasani) [reply src/Microsoft.NetCore.Analyzers/Microsoft.NetCore.Analyzers.md]: I think I figured it out. [GetResourceString](https://github.com/dotnet/roslyn-analyzers/blob/020a095feae4881ea5a13f8ed40c56e478f6a67a/src/Microsoft.NetCore.Analyzers/Core/Security/Helpers/SecurityHelpers.cs#L79) helper method is only expecting resource strings from `MicrosoftNetCoreSecurityResources`, but rules deriving from `UseXmlReaderBase` are passing in resource strings from `SystemSecurityCryptographyResources`. @dotpaul, you want me to file a bug for you to fix this?

Comment (mavasani) [reply src/Microsoft.NetCore.Analyzers/Microsoft.NetCore.Analyzers.md]: Also, @dotpaul if you don't mind, I would still like to go ahead with this PR and you can fix this as a follow up. Just run CIBuild locally to regenerate this file after your fix.

Comment (dotpaul) [reply src/Microsoft.NetCore.Analyzers/Microsoft.NetCore.Analyzers.md]: File an issue for @LingxiaChen, please

Comment (mavasani) [reply src/Microsoft.NetCore.Analyzers/Microsoft.NetCore.Analyzers.md]: https://github.com/dotnet/roslyn-analyzers/issues/2421

**Pull request merged**